freeipa/src/main/java/com/sequenceiq/freeipa/service/image/FreeIpaImageProvider.java

```diff
@@ -53,24 +51,24 @@ public class FreeIpaImageProvider implements ImageProvider {
     private ProviderSpecificImageFilter providerSpecificImageFilter;
 
     @Override
-    public Optional<ImageWrapper> getImage(ImageSettingsRequest imageSettings, String region, String platform) {
-        String imageId = imageSettings.getId();
-        String catalogUrl = StringUtils.isNotBlank(imageSettings.getCatalog()) ? imageSettings.getCatalog() : defaultCatalogUrl;
-        String imageOs = StringUtils.isNotBlank(imageSettings.getOs()) ? imageSettings.getOs() : defaultOs;
+    public Optional<ImageWrapper> getImage(FreeIpaImageFilterSettings imageFilterParams) {
+        String catalogUrl = StringUtils.isNotBlank(imageFilterParams.getCatalog()) ? imageFilterParams.getCatalog() : defaultCatalogUrl;
+        imageFilterParams.setOs(StringUtils.isNotBlank(imageFilterParams.getOs()) ? imageFilterParams.getOs() : defaultOs);
+        imageFilterParams.setCatalog(catalogUrl);
 
         ImageCatalog cachedImageCatalog = imageCatalogProvider.getImageCatalog(catalogUrl);
-        return findImageForAppVersion(region, platform, imageId, imageOs, cachedImageCatalog)
-                .or(() -> retryAfterEvictingCache(region, platform, imageId, catalogUrl, imageOs))
+        return findImageForAppVersion(imageFilterParams, cachedImageCatalog)
+                .or(() -> retryAfterEvictingCache(imageFilterParams))
                 .map(i -> new ImageWrapper(i, catalogUrl, null));
     }
 
-    public List<ImageWrapper> getImages(ImageSettingsRequest imageSettings, String region, String platform) {
-        String imageId = imageSettings.getId();
-        String catalogUrl = StringUtils.isNotBlank(imageSettings.getCatalog()) ? imageSettings.getCatalog() : defaultCatalogUrl;
-        String imageOs = StringUtils.isNotBlank(imageSettings.getOs()) ? imageSettings.getOs() : defaultOs;
+    @Override
+    public List<ImageWrapper> getImages(FreeIpaImageFilterSettings imageFilterSettings) {
+        imageFilterSettings.setCatalog(StringUtils.isNotBlank(imageFilterSettings.getCatalog()) ? imageFilterSettings.getCatalog() : defaultCatalogUrl);
+        imageFilterSettings.setOs(StringUtils.isNotBlank(imageFilterSettings.getOs()) ? imageFilterSettings.getOs() : defaultOs);
```

Comment:
I would prefer creating a new `FreeIpaImageFilterSettings` here instead of updating the one passed as a parameter. It would make the code more readable